### PR TITLE
add dockfile for ROCm 4.2

### PR DESCRIPTION
**Pull request by weixingzhang**
**Description**: Describe your changes.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: orttraining/tools/amdgpu/Dockerfile.rocm4.2.pytorch [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+WORKDIR $GITHUB_DIR
+RUN git clone https://github.com/microsoft/huggingface-transformers.git
+
+ENV BERT_DIR=${WORKSPACE_DIR}/BERT
+ENV GPT2_DIR=${WORKSPACE_DIR}/GPT2
+ENV TRAIN_FILE=${WORKSPACE_DIR}/GPT2/${GPT2_DATASET}/wiki.train.tokens
+ENV TEST_FILE=${WORKSPACE_DIR}/GPT2/${GPT2_DATASET}/wiki.test.tokens
+
+# Enable ssh access without password needed
+RUN sed -i 's/#PermitRootLogin prohibit-password/PermitRootLogin yes/g' /etc/ssh/sshd_config
```
Comment (snnn) [inline orttraining/tools/amdgpu/Dockerfile.rocm4.2.pytorch:148]: Are these four lines needed?

Comment (weixingzhang) [reply orttraining/tools/amdgpu/Dockerfile.rocm4.2.pytorch:144]: yeah, they should be removed as well. Thanks!

Comment (snnn) [reply orttraining/tools/amdgpu/Dockerfile.rocm4.2.pytorch:144]: But it is not changed?

Comment (weixingzhang) [reply orttraining/tools/amdgpu/Dockerfile.rocm4.2.pytorch:144]: Ah, I was thinking you were referring the four lines below. Yeah, I was told that these four lines are needed.

+ENV BERT_DIR=${WORKSPACE_DIR}/BERT
+ENV GPT2_DIR=${WORKSPACE_DIR}/GPT2
+ENV TRAIN_FILE=${WORKSPACE_DIR}/GPT2/${GPT2_DATASET}/wiki.train.tokens
+ENV TEST_FILE=${WORKSPACE_DIR}/GPT2/${GPT2_DATASET}/wiki.test.tokens

**Comment by suffiank** [comment]
Can we remove Dockerfile for Rocm < 4.0?

**Review by suffiank — APPROVED** [review]

**Comment by weixingzhang** [comment]
> Can we remove Dockerfile for Rocm < 4.0?

agree.

**Pull request merged**